File: course/app/src/main/java/com/txwstudio/app/timetable/Ui/Fragments/TimePickerFragment.java
package com.txwstudio.app.timetable.Ui.Fragments;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class TimePickerFragment extends DialogFragment{
//  implements TimePickerDialog.OnTimeSetListener
    // https://www.youtube.com/watch?v=QMwaNN_aM3U

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Calendar c = Calendar.getInstance();
        int hour = c.get(Calendar.HOUR_OF_DAY);
        int min = c.get(Calendar.MINUTE);

        return new TimePickerDialog(getActivity(),
                (TimePickerDialog.OnTimeSetListener) getActivity(),
                hour, min, android.text.format.DateFormat.is24HourFormat(getActivity()));
    }
}

File: course/app/src/main/java/com/txwstudio/app/timetable/Ui/timepicker/TimePickerFragment.kt
package com.txwstudio.app.timetable.ui.timepicker

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.text.format.DateFormat
import androidx.fragment.app.DialogFragment
import java.util.*

class TimePickerFragment : DialogFragment() {
    //  implements TimePickerDialog.OnTimeSetListener
    // https://www.youtube.com/watch?v=QMwaNN_aM3U

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val c = Calendar.getInstance()
        val hour = c[Calendar.HOUR_OF_DAY]
        val min = c[Calendar.MINUTE]
        return TimePickerDialog(activity,
                activity as OnTimeSetListener?,
                hour, min, DateFormat.is24HourFormat(activity))
    }
}